Core/Security/X509Chain.py

```diff
@@ -147,34 +167,33 @@ def loadProxyFromString( self, pemData ):
     retVal = self.loadChainFromString( pemData )
     if not retVal[ 'OK' ]:
       return retVal
-    return self.loadKeyFromString( pemData )
+    return self.loadKeyFromString( pemData, M2Crypto.util.no_passphrase_callback )
 
   def __getProxyExtensionList( self, diracGroup = False, rfc = False, rfcLimited = False ):
     """
     Get the list of extensions for a proxy
     """
-    extList = []
-    extList.append( crypto.X509Extension( 'keyUsage',
-                                          'critical, digitalSignature, keyEncipherment, dataEncipherment' ) )
+    extStack = M2Crypto.X509.X509_Extension_Stack()
+    kUext = M2Crypto.X509.new_extension( 'keyUsage',
+                                          'digitalSignature, keyEncipherment, dataEncipherment', critical = 1 )
+    extStack.push( kUext )
     if diracGroup and type( diracGroup ) in self.__validExtensionValueTypes:
-      extList.append( crypto.X509Extension( 'diracGroup', diracGroup ) )
+      dGext = M2Crypto.X509.new_extension( 'diracGroup', diracGroup)
+      extStack.push( dGext )
     if rfc or rfcLimited:
-      blob = [ [ "1.3.6.1.5.5.7.21.1" ] ] if not rfcLimited else [ [ "1.3.6.1.4.1.3536.1.1.1.9" ] ]
-      asn1Obj = crypto.ASN1( blob )
-      asn1Obj[0][0].convert_to_object()
-      asn1dump = binascii.hexlify( asn1Obj.dump() )
-      extval = "critical,DER:" + ":".join( asn1dump[i:i + 2] for i in range( 0, len( asn1dump ), 2 ) )
-      ext = crypto.X509Extension( "proxyCertInfo", extval )
-      extList.append( ext )
-    return extList
+      if rfc:
+        ext =  M2Crypto.X509.new_extension( 'proxyCertInfo', 'critical, language:1.3.6.1.5.5.7.21.1', critical = 1 )
```

Comment:
I would love to see an explanation of this.

And in general, I would love to see some documentation. Would you provide, please? before we (and especially you) forget what's this all about.